### PR TITLE
Add watcher-tempest-plugin to stable/caracal

### PR DESCRIPTION
**Pull request by hemanthnakkina**
Add watcher-tempest-plugin to stable/caracal

commit 550e43bb45 in python-temepstconf introduces
discovery of watcher service and updates tempest
config accordingly. The changes from v3.5.0 to the
above mentioned commit are few that should not break
anything on caracal. The changes introduced are case
sensitive for tempest config options and discovery
of watcher service. So install python-tempestconf to
the above commit in the snap to include watcher
discoverability feature.
    
Add excluded-plugins.txt from snap-tempest-automation
project and remove watcher-tempest-plugin from the list.

Depends-On:
- [x] https://review.opendev.org/c/openinfra/python-tempestconf/+/927893

### CODE REVIEW

**Review by chanchiwai-ray — COMMENTED** [review]
If it's useful for other releases, maybe it's better to remove that from the exclude list: https://github.com/canonical/snap-tempest-automation/blob/main/excluded-plugins.txt#L22C1-L22C23. Then the automation will generate a new `snapcraft.yaml` to all release branch

**Review by samuelallan72 — DISMISSED** [review]
Sorry the dependencies in snapcraft.yaml are controlled by the automation repo as @chanchiwai-ray noted, so they shouldn't be modified in-place.

I believe to modify the excluded-plugins for a particular branch, you can put an excluded-plugins.txt in the root of this repository in the branch you want (stable/caracal), and the automation will update snapcraft.yaml accordingly.

I think it's fine to update snapcraft.yaml too, but you definitely need the config for the automation as well, otherwise it will immediately want to overwrite it. :)

In concrete terms, I think that means there are two possible ways you could go here:

1. if watcher-tempest-plugin should be enabled for all tempest openstack releases, then this PR can be dropped and excluded-plugins.txt can be updated in snap-tempest-automation.
2. if only should be enabled for this branch, then you can copy the excluded-plugins.txt from snap-tempest-automation into here, and remove the line containing watcher-tempest-plugin. :)

**Comment by aieri** [comment]
@hemanthnakkina this link should also help explain how the automation works: https://github.com/canonical/snap-tempest/blob/main/CONTRIBUTING.md

**Comment by hemanthnakkina** [comment]
I see from [1] the watcher service by default is considered as enable from the watcher-tempest-plugin.
discover-tempest-config does not look for watcher service and wont do any updates in tempest conf.
That is the reason watcher is in excluded-plugins.txt

Is it ok if i remove `watcher` from `excluded-plugins` for caracal release following suggestions by samuel and adrea?
This also means tempest-k8s charm should use `--append` option [2] to update tempest.conf to include/exclude watcher service.
Are there any other tools that uses snap-tempest since they might also require updates if watcher-tempest-plugin is part of the snap?

[1] https://opendev.org/openstack/watcher-tempest-plugin/src/tag/3.1.0/watcher_tempest_plugin/config.py#L21
[2] https://docs.opendev.org/openinfra/python-tempestconf/latest/cli/cli_options.html

**Comment by samuelallan72** [comment]
@hemanthnakkina thanks, I think that sounds ok to me.  @aieri and @chanchiwai-ray have more context on the original reasons these plugins were disabled here, and may be able to confirm. :)

**Comment by aieri** [comment]
Some context: the exclude list was added in #133 because otherwise test runs would return failed tests for endpoints that aren't available (unless the user knows enough about tempest to manually customize their config file). And indeed  tempestconf is unable to detect the absence of watcher (see https://storyboard.openstack.org/#!/story/2011096), although at least the plugin seems to respect the config file.

@hemanthnakkina given the objective of snap-tempest being an easy to use tempest package, adding the watcher plugin without fixing tempestconf first would cause regressions in environments that don't have watcher. What is the end goal of this change?

**Comment by hemanthnakkina** [comment]
> Some context: the exclude list was added in #133 because otherwise test runs would return failed tests for endpoints that aren't available (unless the user knows enough about tempest to manually customize their config file). And indeed tempestconf is unable to detect the absence of watcher (see https://storyboard.openstack.org/#!/story/2011096), although at least the plugin seems to respect the config file.
> 
> @hemanthnakkina given the objective of snap-tempest being an easy to use tempest package, adding the watcher plugin without fixing tempestconf first would cause regressions in environments that don't have watcher. What is the end goal of this change?

We are introducing watcher-k8s charm in sunbeam and so looking at tempest-tests

**Comment by hemanthnakkina** [comment]
I submitted https://review.opendev.org/c/openinfra/python-tempestconf/+/927893 to fix in upstream.

**Comment by aieri** [comment]
> I am planning to submit a PR to python-tempestconf but doubt if the change will be backported to the version used in snap-tempest stable/caracal.

tempestconf does not have per-OpenStack-versions so our automation picks up whichever is the latest one. No backport will be necessary. See https://github.com/canonical/snap-tempest-automation for more details.

File: snap/snapcraft.yaml
Comment (samuelallan72) [inline snap/snapcraft.yaml:65]: I'm not sure how this will interact with the automation, but fine to include it now.  We can update automation later. :)

Note also that the automation will clobber the comment here.

**Review by samuelallan72 — APPROVED** [review]

File: snap/snapcraft.yaml
Comment (hemanthnakkina) [reply snap/snapcraft.yaml:65]: I see your point. What are your thoughts on automation updates required for these kind of changes?

Comment (samuelallan72) [reply snap/snapcraft.yaml:65]: I'm not sure - I think this is fine for now - automation appears to pull the latest release of python-tempestconf.  So we can just hold off automated updates until this is released?

**Review by chanchiwai-ray — APPROVED** [review]
I think we can land this first, and go back to fix it properly when the tempest conf is updated

File: snap/snapcraft.yaml
Comment (chanchiwai-ray) [reply snap/snapcraft.yaml:65]: @samuelallan72 do you have more any concern?

Comment (hemanthnakkina) [reply snap/snapcraft.yaml:65]: @chanchiwai-ray Can you hold merging the PR as I am discussing with @aieri on better approaches to handle these kind of scenarios. I will bring the discussion in this PR once we have some direction

Comment (samuelallan72) [reply snap/snapcraft.yaml:65]: @chanchiwai-ray no concerns from me, but agreed that we should hold off until we get the go-ahead from Andrea and Hemanth. :)

**Review by aieri — APPROVED** [review]
I think the automation will try to overwrite this but this is fine for a temporary fix

File: snap/snapcraft.yaml
Comment (hemanthnakkina) [reply snap/snapcraft.yaml:65]: I discussed with Andrea and we will merge this PR for now but in future will try to update automation to have manual-requirements have precedence over plugin requirements or tempest conf requirements.

@samuelallan72 Can you resolve the conversation and merge the PR, thanks

**Comment by chanchiwai-ray** [comment]
@hemanthnakkina side question, when your patch to python-tempesf conf is landed and released, does that means the [concern](https://github.com/canonical/snap-tempest/pull/226#issuecomment-2327861775) from @aieri can be resolved? and we can always add watcher plugin (even for other openstack release)

**Comment by hemanthnakkina** [comment]
> on-tempestconf is landed and r

Yes since all the openstack releases seems to be pointing to the latest tempestconf.

**Pull request merged**
